public/components/trace_analytics/components/common/__tests__/legacy_route_helpers.test.tsx

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  convertLegacyTraceAnalyticsUrl,
+  convertTraceAnalyticsNewNavUrl,
+} from '../legacy_route_helpers';
+import {
+  observabilityTracesID,
+  observabilityTracesNewNavID,
+  observabilityServicesNewNavID,
+} from '../../../../../../common/constants/shared';
+import { coreRefs } from '../../../../../framework/core_refs';
+
+describe('ConvertLegacyTraceAnalyticsUrl', () => {
+  it('should convert legacy URL correctly', () => {
+    const location = {
+      pathname: '/app/trace-analytics-dashboards',
+      hash: '#/traces',
+      search: '?param1=value1',
+    } as Location;
+
+    const result = convertLegacyTraceAnalyticsUrl(location);
+    expect(result).toBe(`/app/${observabilityTracesID}#/traces?param1=value1`);
+  });
```

Comment:
Can we add more test cases for the legacy URL redirection with tracesId and servicename. E.g of lagacy trace URL with traceID: https://playground.opensearch.org/app/trace-analytics-dashboards#/traces/02feb3a4f611abd81f2a53244d1278ae

Reply:
Added four test to cover these conditions for new navigation on and off.